vignettes/articles/writing-custom-fns.Rmd

```diff
@@ -0,0 +1,244 @@
+
+
+```{r child="children/_custom-fn-available-args.Rmd", echo=FALSE, results="asis"}
+```
+
+### Additional arguments
+
+You can add additional arguments to custom check functions and pass values to them by including them in the `args` configuration in the `validations.yml` file. These values are passed to the custom check function by `hubValidations` when the function is called.
+
+
+If you do add additional arguments to a custom check function, you should also add input  checks at the start of the function to ensure inputs are valid. Package `checkmate` contains a wide range of functions for checking inputs.
+
+For example, the optional check `opt_check_tbl_col_timediff()` (which is deployed in exactly the same fashion as custom functions, i.e. through the `validations.yml` file) takes additional arguments `t0_colname`, `t1_colname` and `timediff`. 
+
+
+```{r}
+opt_check_tbl_col_timediff
+```
+
+
+
+You can add an example extra argument with `extra_args = TRUE` when creating the custom check function with `create_custom_check()`.
+
+```{r}
+create_custom_check("cstm_check_tbl_args",
+  hub_path = hub_path,
+  extra_args = TRUE
+)
+```
+
+This adds an extra argument `extra_arg` to the custom check function formals as well as an example input check.
+s
+```{r, code=readLines(r_path(hub_path, "cstm_check_tbl_args")), eval=FALSE}
+```
+
+
+## Function output
+
+### Capturing and returning check results
+
+**Returning a check condition with `capture_check_cnd()`**
+
+The `capture_check_cnd()` function is used to return a check condition and it's output should also be what a custom check function returns in most cases. The function returns a `<hub_check>` class object depending on the value passed to the `check` argument, which represents the summary of the condition being checked by a given validation function. 
+
+If the value passed to `check ` is `TRUE`, the function returns a `<message/check_success>` class object. 
+
+If the value is `FALSE`, the output depends on the `error` argument.  
+
+- If `error` is `TRUE`, the function returns a `<message/check_error>` class object. If `error` is `FALSE`, the function returns a `<error/check_failure>` class object, which indicates the check has failed. 
+- If `error` is `TRUE`, the function returns a `<error/check_error>` class object, which indicates the check has failed and additionally causes execution of further custom validation functions to halt. Set `error = TRUE` if downstream checks cannot be run if the current check fails.
+
+The default value of `error` is `FALSE` in `create_custom_check()` but setting it to `TRUE` sets the error argument to `TRUE`.
+
+```{r}
+create_custom_check("cstm_check_tbl_error",
+  hub_path = hub_path, error = TRUE
+)
+```
+
+```{r, code=readLines(r_path(hub_path, "cstm_check_tbl_error")), eval=FALSE}
+```
+
+
+### Skipping a check
+
+**Returning a check message with `capture_check_info()`**
```

Comment:
```suggestion
```

Reply:
I don't really want it as a sub heading as it's not adding anything to the organisation tree of the document. It's more of a subtitle for that particular section.

I think it looks fine in the rendered version?
<img width="819" alt="image" src="https://github.com/user-attachments/assets/31ec3950-684c-4768-b23a-69e7bfe2162a">

There's also exactly the same in the previous section:

<img width="830" alt="image" src="https://github.com/user-attachments/assets/7db013ca-7b75-4292-b3e6-053f8dfd24c3">